com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs

```diff
@@ -3506,6 +3475,60 @@ ShadowMapType shadowMapType
             ? ShadowMapType.PunctualAtlas
             : ShadowMapType.CascadedDirectional;
 
+
+        internal void UpdateRenderEntity()
+        {
+            HDLightEntityCollection lightEntities = HDLightEntityCollection.instance;
+            ref HDLightRenderData lightRenderData = ref lightEntities.GetLightData(lightEntity);
+            lightRenderData.pointLightType = m_PointlightHDType;
+            lightRenderData.spotLightShape = m_SpotLightShape;
+            lightRenderData.areaLightShape = m_AreaLightShape;
+            lightRenderData.lightLayer = m_LightLayers;
+            lightRenderData.fadeDistance = m_FadeDistance;
+            lightRenderData.distance = m_Distance;
+            lightRenderData.angularDiameter = m_AngularDiameter;
+            lightRenderData.volumetricFadeDistance = m_VolumetricFadeDistance;
+            lightRenderData.includeForRayTracing = m_IncludeForRayTracing;
+            lightRenderData.useScreenSpaceShadows = m_UseScreenSpaceShadows;
+            lightRenderData.useRayTracedShadows = m_UseRayTracedShadows;
+            lightRenderData.colorShadow = m_ColorShadow;
+            lightRenderData.lightDimmer = m_LightDimmer;
+            lightRenderData.volumetricDimmer = m_VolumetricDimmer;
+            lightRenderData.shadowDimmer = m_ShadowDimmer;
+            lightRenderData.shadowFadeDistance = m_ShadowFadeDistance;
+            lightRenderData.volumetricShadowDimmer = m_VolumetricShadowDimmer;
+            lightRenderData.shapeWidth = m_ShapeWidth;
+            lightRenderData.shapeHeight = m_ShapeHeight;
+            lightRenderData.flareSize = m_FlareSize;
+            lightRenderData.flareFalloff = m_FlareFalloff;
+            lightRenderData.aspectRatio = m_AspectRatio;
+            lightRenderData.innerSpotPercent = m_InnerSpotPercent;
+            lightRenderData.spotIESCutoffPercent = m_SpotIESCutoffPercent;
+            lightRenderData.shapeRadius = m_ShapeRadius;
+            lightRenderData.barnDoorLength = m_BarnDoorLength;
+            lightRenderData.barnDoorAngle = m_BarnDoorAngle;
+            lightRenderData.affectDiffuse = m_AffectDiffuse;
+            lightRenderData.affectSpecular = m_AffectSpecular;
+            lightRenderData.applyRangeAttenuation = m_ApplyRangeAttenuation;
+            lightRenderData.penumbraTint = m_PenumbraTint;
+            lightRenderData.interactsWithSky = m_InteractsWithSky;
+            lightRenderData.surfaceTint = m_SurfaceTint;
+            lightRenderData.shadowTint = m_ShadowTint;
+            lightRenderData.flareTint = m_FlareTint;
+        }
```

Comment:
If we add another field, I'm pretty sure we will forget to update the list here as we forgot to amend the Copy in the past.
You should add a test on that. It is not simple but I think you can do it the same way the Copy is tested (using the attributes and some reflection, see Editor tests)

Reply:
This one is interesting, since these fields shouldnt be read back by anything but the rendering pipeline in burst. Adding a field here arbitrarily could potentially hurt performance (since this is what is read on the burst side of things). I will add a comment explaining this. Basically, if a field is not used in the GPU, this should not be added for a field.

Thus, I think I should:
* Add a test indeed that checks the state of the GPU side array
* Add a comment explaining what to do, and when to add a field.
* Perhaps rename the HDLightCollection to something like HDLightRenderDatabase or HDLightGpuDatabase. Something along those lines.

Let me know your thoughts. I left more comments in slack.